pandas/core/groupby/numba_.py

```diff
@@ -0,0 +1,138 @@
+import inspect
+import types
+
+import numpy as np
+
+from pandas.compat._optional import import_optional_dependency
+
+
+class InvalidApply(Exception):
```

Comment:
why do you need a new exception?

Reply:
Sure will want to import this from it's original location in cython. Or we can move `InvalidApply` to `pandas.errors`